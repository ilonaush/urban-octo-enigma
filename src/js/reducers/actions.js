export default {
    employWorker: function (worker) {
        return {
            type: 'EMPLOY_WORKER',
            worker: worker
        }
    },
    editWorkTime: function (worker) {
        return {
            type: 'EDIT_WORK_TIME',
            worker: worker,
        }
    },
    fireWorker: function (workerId) {
        return {
            type: 'FIRE_WORKER',
            id: workerId
        }
    }
};
