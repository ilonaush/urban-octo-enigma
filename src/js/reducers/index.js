import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer   from './reducers';
import  { saveStore, loadStore } from '../services/LocalStorageService';

const predefinedStore = loadStore() || {};

const store = createStore(
    reducer,
    predefinedStore

);

store.subscribe(() => {
    saveStore(store)
});

export {
    store, predefinedStore
};
