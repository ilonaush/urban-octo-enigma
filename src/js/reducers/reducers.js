import {ACTIONS_TYPES} from "../constants";

const initialState = {
  loading: false,
};

let reducer = (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS_TYPES.GET_CATS:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ADD_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.FEED_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.HUG_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.WASH_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ISSUE_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.SET_LOADING: {
      return {
        ...state,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      }
    }
    default:
      return state;
  }
};

export default reducer;
