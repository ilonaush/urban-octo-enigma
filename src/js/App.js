import React from 'react';
import { Provider } from 'react-redux';
import Layout from "./components/Layout/Layout";
import {store} from "./reducers/index";
import actions from './reducers/actions';


export default class App extends React.Component{
    constructor(props) {
        super(props);
        store.dispatch(actions.getWorkers());
    }
    render() {
        return (
            <Provider store={store}>
                <Layout/>
            </Provider>
        )
    }
}


