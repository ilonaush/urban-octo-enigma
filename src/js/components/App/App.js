import React from 'react';
import {Provider} from 'react-redux';
import Layout from "../Layout/Layout";
import {store} from "../../reducers/index";
import actions from '../../reducers/actions';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    store.dispatch(actions.getCats());
  }

  render() {
    return (
        <Provider store={store}>
          <Layout/>
        </Provider>
    )
  }
}


