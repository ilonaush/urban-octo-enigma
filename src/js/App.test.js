import React from 'react';

import App  from './App';

import ReactDOM from 'react-dom';
import {MemoryRouter} from "react-router";
import {Provider} from "react-redux";
import {store} from "./reducers/index";


describe('App', () => {
    it('renders', () => {

        const div = document.createElement('div');
        ReactDOM.render(<MemoryRouter><Provider store={store}><App/></Provider></MemoryRouter>, div)
    })
})