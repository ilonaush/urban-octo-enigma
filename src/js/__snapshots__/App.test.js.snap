// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="layout"
>
  <div
    className="sidebar"
  >
    <a
      className="heading"
      href="/"
      onClick={[Function]}
    >
      Dashboard
    </a>
    <hr />
    <div>
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            <span
              alt=""
              className="icon"
              style={
                Object {
                  "backgroundImage": "url('images/list.svg",
                }
              }
            />
            Home
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/add-worker"
            onClick={[Function]}
          >
            <span
              alt=""
              className="icon"
              style={
                Object {
                  "backgroundImage": "url('images/add.svg",
                }
              }
            />
            Add a worker
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/fire-worker"
            onClick={[Function]}
          >
            <span
              alt=""
              className="icon"
              style={
                Object {
                  "backgroundImage": "url('images/fire.svg",
                }
              }
            />
            Fire worker
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/gallery"
            onClick={[Function]}
          >
            <span
              alt=""
              className="icon"
              style={
                Object {
                  "backgroundImage": "url('images/gallery.svg",
                }
              }
            />
            Gallery
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="dashboard"
  >
    <div
      className="intro"
    >
      Table
    </div>
    <div
      className="d-content"
    />
  </div>
</div>
`;
